<launch>

    <arg name="camera" default="camera" />

    <node pkg="rviz" type="rviz" name="rviz" />
    <!--<include file="$(find realsense_tools)/launch/rs_camera.launch" />
    -->

    <arg name="yolo_model" default="$(find object_detection)/cfg/yolo_detection.yaml" />
    <arg name="sort_tracking" default="$(find object_detection)/cfg/sort_tracking.yaml" />

    <node pkg="object_detection" type="yolo_detection.py" name="Yolo" output="screen">

        <remap from="image" to="/camera/color/image_raw" />
        <remap from="output_image" to="/detection/yolo/objects_image" />
        <remap from="bounding_boxes" to="/detection/yolo/bboxes" />
        <remap from="camera_info" to="/camera/aligned_depth_to_color/camera_info" />
        <remap from="depth_image" to="/camera/aligned_depth_to_color/image_raw" />

        <rosparam command="load" file="$(arg yolo_model)" />

    </node>

    <node pkg="object_detection" type="sort_tracking.py" name="SORT" output="screen">

        <remap from="image" to="/camera/color/image_raw" />
        <remap from="bounding_boxes" to="/detection/yolo/bboxes" />
        <remap from="camera_info" to="/camera/aligned_depth_to_color/camera_info" />
        <remap from="depth_image" to="/camera/aligned_depth_to_color/image_raw" />

        <remap from="output_image" to="/tracking/yolo/objects_image" />
        <remap from="object" to="/tracking/yolo/object" />

        <rosparam command="load" file="$(arg sort_tracking)" />

    </node>

</launch>